build-scripts/generate_bash_scripts.py

```diff
@@ -0,0 +1,258 @@
+        value_id = refine_value.get("idref")
+        selector = refine_value.get("selector")
+        refinements[value_id] = selector
+    return refinements
+
+
+def get_remediation(rule):
+    for fix in rule.findall("./{%s}fix" % XCCDF12_NS):
+        if fix.get("system") == "urn:xccdf:fix:script:sh":
+            return fix
+    return None
+
+
+def _get_all_itms(benchmark, element_name, callback):
+    itms = {}
+    el_xpath = ".//{%s}%s" % (XCCDF12_NS, element_name)
+    for el in benchmark.findall(el_xpath):
+        rule_id = el.get("id")
+        itms[rule_id] = callback(el)
+    return itms
+
+
+def get_all_remediations(benchmark):
+    return _get_all_itms(benchmark, "Rule", get_remediation)
+
+
+def get_variable_values(variable):
+    values = {}
+    for value in variable.findall("./{%s}value" % (XCCDF12_NS)):
+        selector = value.get("selector")
+        if selector is None:
+            selector = DEFAULT_SELECTOR
+        if value.text is None:
+            values["selector"] = ""
+        else:
+            values[selector] = value.text
+    return values
+
+
+def get_all_variables(benchmark):
+    return _get_all_itms(benchmark, "Value", get_variable_values)
+
+
+def expand_variables(fix_el, refinements, variables):
+    content = fix_el.text[:]
+    for sub_el in fix_el.findall("./{%s}sub" % XCCDF12_NS):
+        variable_id = sub_el.get("idref")
+        values = variables[variable_id]
+        selector = refinements.get(variable_id, DEFAULT_SELECTOR)
+        if selector == "default":
+            return None
+        if selector in values:
+            value = values[selector]
+        else:
+            value = list(values.values())[0]
+        content += value
+        content += sub_el.tail
+    return content
+
+
+def create_header(benchmark_data, profile):
+    shebang_with_newline = "#!/usr/bin/env bash\n"
+    remediation_type = "Bash Remediation Script"
+    profile_title = profile.find("./{%s}title" % XCCDF12_NS).text
+    description = profile.find("./{%s}description" % XCCDF12_NS).text
+    commented_profile_description = ""
+    for line in description.split("\n"):
+        trimmed_line = line.strip()
+        if trimmed_line != "":
+            commented_line = "# " + trimmed_line + "\n"
+            commented_profile_description += commented_line
+    profile_id = profile.get("id")
+    xccdf_version_name = "1.2"
+    oscap_version = "1.3.8"
+    profile_id = profile.get("id")
+    fix_type = "bash"
+    remediation_type = "Bash Remediation Script"
+    how_to_apply = "$ sudo ./remediation-script.sh"
+    fix_header = (
+        "%s"
+        "%s\n"
+        "#\n"
+        "# %s for %s\n"
+        "#\n"
+        "# Profile Description:\n"
+        "%s"
+        "#\n"
+        "# Profile ID:  %s\n"
+        "# Benchmark ID:  %s\n"
+        "# Benchmark Version:  %s\n"
+        "# XCCDF Version:  %s\n"
+        "#\n"
+        "# This file was generated by OpenSCAP %s using:\n"
```

Comment:
I'm not sure how I feel about this. This is not true.

 Maybe we want to change "can be generated with"

Reply:
I intentionally left it there to keep it identical and I was wondering if there will be any opinion on it.  I agree with you,  I will improve it in the next commit.